lib/ansible/modules/system/firewalld_ipset.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, [rwaweber]
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: firewalld_ipset
+
+short_description: A module to manage firewalld ipsets
+
+description:
+  - This module utilizes the firewalld python client to manipulate and manage
+    the existence of ipsets. These can be ipv4, ipv6, and mac addresses. It is
+    particularly useful if you take advantage of the firewalld zone model.
+
+version_added: "2.8"
+
+author: Will Weber (@rwaweber)
+
+options:
+  name:
+    description: A name for the given ipset
+    required: True
+    type: str
+  settype:
+    description:
+      - A type declaration for the given set of addresses. Supported types can
+        be found in the firewalld code
+        https://github.com/firewalld/firewalld/blob/master/src/firewall/core/ipset.py#L36.
+        Typical usage with ips or mac addresses will use either a "hash:ip"
+        or "hash:mac" option.
+    required: True
+    choices:
+      - "hash:ip"
+      - "hash:ip,port"
+      - "hash:ip,port,ip"
+      - "hash:ip,port,net"
+      - "hash:ip,mark"
+      - "hash:net"
+      - "hash:net,net"
+      - "hash:net,port"
+      - "hash:net,port,net"
+      - "hash:net,iface"
+      - "hash:mac"
+  state:
+    description: The desired state of the ipset
+    required: True
+    choices: [ "present", "absent" ]
+  permanent:
+    description: Should the changes be applied permanently
+    required: False
+    default: False
+    type: bool
+  immediate:
+    description: Should the changes be applied immediately
+    required: False
+    default: False
+    type: bool
+  addresses:
+    description: The list of addresses to be added or removed from the ipset. This can be ipv4, ipv6, and mac addresses.
+    required: False
+    type: list
```

Comment:
Is there any desire to support offline firewall operations like the `firewalld` module does?